products/storage/terraform.yaml

```diff
@@ -102,6 +102,22 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         ignore_read: true
   Object: !ruby/object:Overrides::Terraform::ResourceOverride
     exclude: true
+  HmacKey: !ruby/object:Overrides::Terraform::ResourceOverride
```

Comment:
Can you add `exclude_import: true` to the resource? The secret is irretrievable, so I'd prefer we omit it.

However, you're more familiar with the product than me- would there be a case that users would want to manage already-created keys in Terraform even without the secret, such as to update the state?

Reply:
Does `exclude_import` apply only to the importing of an already created key into the statefile? ie: you can still modify the hmac key "state" of a key if you created it in tf? 
Regardless, I feel like I've been the person that curses the one who didn't create a data piece (datum?) to go with the resource and had to do statefile surgery for _reasons_.  Speaking of which, it looks like that is not being autogenerated, so is my choice to `exclude_import` or to create a data bit?

Reply:
> Does exclude_import apply only to the importing of an already created key into the statefile? ie: you can still modify the hmac key "state" of a key if you created it in tf?

That's correct, `exclude_import` will remove the ability to `terraform import` the resource. We would have no way of setting the secret on an imported resource.

I'm not sure what you mean by a data bit- a datasource? If that's the case, they're handwritten but based on the schema/read of the resource proper: https://github.com/terraform-providers/terraform-provider-google/blob/master/google/data_source_google_compute_router.go

A datasource would have the same restriction, that `secret` is unrecoverable, but would let the user see the `state` of a key.

Reply:
Cool, thanks for the explanation. Yeah, we would never want to change the secret, but we might want to import it and activate/deactivate it.
I got my terraform importing vs referencing it jargon/workflow mixed up, but yes, I was reaching for the word datasource and kept finding data resource which wasn't right.